h/util/group_scope.py

```diff
@@ -18,25 +18,73 @@ def match(uri, scopes):
     return scope in scopes
 
 
+def uri_in_scope(uri, scopes):
+    """
+    Does the URI match any of the scope patterns?
+
+    Return True if the URI matches one or more patterns in scopes (if the
+    URI string begins with any of the scope strings)
+
+    :arg uri: URI string in question
+    :arg scopes: List of URIs that define scope
+    :type scopes: list(str)
+    :rtype: bool
+    """
+    scope_matches = [scope for scope in scopes if uri.startswith(scope)]
+
+    if len(scope_matches):
+        return True
+    return False
```

Comment:
I think the idiomatic way to do this in Python is by using the `any` builtin and a list or generator comprehension: `return any((uri.startswith(scope) for scope in scopes))`, which also avoids calling `startswith()` for every scope if the first one matches